### ISSUE TITLE
F.cross_entropy in ACGAN

### ISSUE THREAD

**Issue by IMGW-univie** [nocotan/pytorch-lightning-gans#9]
I've noticed that you apply a Softmax before doing F.cross_entropy. Perhaps you would be interested to know that as per the [docs](https://pytorch.org/docs/stable/nn.functional.html#cross-entropy), this function already applies a Softmax by itself.

**Comment by nocotan** [comment]
@IMGW-univie Thanks for pointing that out!

I agree with you and I think this is a bug.
The correct code may be as follows:

L76:

```python
self.aux_layer = nn.Sequential(nn.Linear(128 * ds_size ** 2, n_classes))
```

Would you be willing to make a pull request to fix this bug?
If not, I will fix this myself as soon as possible.

**Comment by IMGW-univie** [comment]
Sorry, I didn't fork your repository. Just noticed while browsing through. Btw, there you might also use binary_cross_entropy_with_logits for adv layer, as it was shown to be more numerically stable

`def adversarial_loss(self, y_hat, y):
        return F.binary_cross_entropy_with_logits(y_hat, y)`

**Comment by nocotan** [comment]
@IMGW-univie Thank you!
I will fix them!
